lessons/PowerShell/managed-identity-powershell.md

```diff
@@ -0,0 +1,240 @@
+    StatusCode = $StatusCode
+    Body = $body
+})
+```
+> 📝 **Tip** - Watch out – the Graph API will this way return up to 100 groups – Please adjust with query parameters as needed like `https://graph.microsoft.com/v1.0/groups?$top=42` or use pagination, which is described here: [Paging Microsoft Graph data in your app](https://docs.microsoft.com/en-us/graph/paging)
+
+Take a moment to understand what the code does: 
+
+* log that a request was received
+* define the scope (if not stated, it’s Microsoft Graph)
+* obtain the token from the environment variables IDENTITY_ENDPOINT and IDENTITY_HEADER (more about that in the next step!)
+* pass that token in the header of the REST call towards the group endpoint of Microsoft Graph
+* get the status code (hopefully 200) 🤞
+
+## 2. Create Azure resources
+
+Now we want to create all resources that we need in Azure: 
+
+For testing purposes, I pseudo-randomized a number to not always need to come up with new names:
+
+```
+#Get a random number between 100 and 300 to more easily be able to distinguish between several trials
+$rand = Get-Random -Minimum 100 -Maximum 300
+```
+We will now set some variables, this reduces risk of typos and makes our code better readable – also we can reuse it better – this is a courtesy to future-self
+```
+#Set values
+$resourceGroup = "DemoPlay$rand"
+$location = "westeurope"
+$storage = "luisedemostorage$rand"
+$functionapp = "LuiseDemo-functionapp$rand"
+
+```
+Let’s create a resource-group that will later hold our Azure Functions App
+```
+#create group
+az group create -n $resourceGroup -l $location
+```
+
+As our Functions App will need a storage account, we will create this as well:
+
+```
+#create storage account
+az storage account create `
+  -n $storage `
+  -l $location `
+  -g $resourceGroup `
+  --sku Standard_LRS
+```
+Now create the Azure Functions App which later holds our function (remember we created that earlier locally, but will later deploy it to Azure)
+
+```
+#create function
+az functionapp create `
+  -n $functionapp `
+  --storage-account $storage `
+  --consumption-plan-location $location `
+  --runtime powershell `
+  -g $resourceGroup `
+  --functions-version 3
+```
+
+> 🔎 **Observation** - < It will take a few moments for everything to be set, once this step is completed, you will be prompted with a message, that you also can benefit from Application Insights. >
+
+## 3. Create Managed Identity and assign permissions
+
+We want things to be super secure – this is why we want to enable a system assigned Managed Identity for our new function:
+
+### Steps
+
+```
+az functionapp identity assign -n $functionapp -g $resourceGroup
+```
+Our Managed Identity shall have the right permission scope to access Graph API for Group.Read.All, and to eventually be able to make the required REST call, we will need
+
+- the Graph API service Provider
+- permission scope, expressed as App role
+Let’s do this:
+
+```
```

Comment:
```suggestion
```powershell
```

Reply:
❓I did these in azurecli - shall I still change to powershell?

Reply:
✅